proxy/IPAllow.cc

```diff
@@ -81,47 +88,70 @@ IpAllow::reconfigure()
 IpAllow *
 IpAllow::acquire()
 {
-  return (IpAllow *)configProcessor.get(configid);
+  return static_cast<IpAllow *>(configProcessor.get(configid));
+}
+
+void
+IpAllow::release(IpAllow *config)
+{
+  configProcessor.release(configid, config);
 }
 
 void
-IpAllow::release(IpAllow *lookup)
+IpAllow::release()
+{
+  configProcessor.release(configid, this);
+}
+
+auto
+IpAllow::match(sockaddr const *ip, match_key_t key) -> ACL
```

Comment:
To me anyway, replacing auto with IpAllow::ACL would be more clear than the "-> ACL".  The new syntax isn't really more concise.

Reply:
I've been working with this notation, and the rule I was using was to declare the return type normally if it's an outer scope type, and via `auto` if it's a dependent scope type. It's nice in situations where the dependent type is deeply nested, like
```
GenericClass::NestedContainer::iterator::value_type
GenericClass::NestedContainer::iterator::operator * () { ... }
// vs.
auto
GenericClass::NestedContainer::iterator::operator * () -> value_type { ... }
```
I prefer a simple rule to discussions about exactly how many characters are enough to sufficiently concise.